src/System.Drawing.Common/src/System/Drawing/Pen.Unix.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing.Drawing2D;
+using System.Runtime.InteropServices;
+
+namespace System.Drawing
+{
+    public partial class Pen
+    {
+        // libgdiplus does not implement GdipGetPenCustomEndCap, so we cache the last-known value here.
+        // Note that this value is not necessarily in sync with the true native value of this property,
+        // as it could have been set outside of the CustomEndCap property on this type.
+        private CustomLineCap _cachedEndCap;
+
+        /// <summary>
+        /// Gets or sets a custom cap style to use at the beginning of lines drawn with this <see cref='Pen'/>.
+        /// </summary>
+        public CustomLineCap CustomStartCap
+        {
+            get
+            {
+                IntPtr lineCap = IntPtr.Zero;
+                int status = SafeNativeMethods.Gdip.GdipGetPenCustomStartCap(new HandleRef(this, NativePen), out lineCap);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+                if (lineCap == IntPtr.Zero)
+                {
+                    throw new ArgumentException(SR.Format(SR.GdiplusInvalidParameter));
+                }
+
+                return CustomLineCap.CreateCustomLineCapObject(lineCap);
+            }
+            set
+            {
+                if (_immutable)
+                {
+                    throw new ArgumentException(SR.Format(SR.CantChangeImmutableObjects, nameof(Pen)));
+                }
+
+                int status = SafeNativeMethods.Gdip.GdipSetPenCustomStartCap(new HandleRef(this, NativePen),
+                                                              new HandleRef(value, (value == null) ? IntPtr.Zero : value.nativeCap));
+                SafeNativeMethods.Gdip.CheckStatus(status);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a custom cap style to use at the end of lines drawn with this <see cref='Pen'/>.
+        /// </summary>
+        public CustomLineCap CustomEndCap
+        {
+            get
+            {
+                // If the CustomEndCap has never been set, this accessor should throw.
+                if (_cachedEndCap == null)
+                {
+                    throw new ArgumentException(SR.Format(SR.GdiplusInvalidParameter));
```

Comment:
In this case, it might help to think of the native function itself, which does have a parameter (the native handle of the pen). Still, it's not much of an "argument exception" (IMO), because the parameter passed in is actually a valid Pen handle -- it just doesn't have an end cap associated with it.